### PR TITLE
[V2 API] Migrate place stat var union/existence and place name related APIs

### PR DESCRIPTION
**Pull request by shifucun**
* Use /info/place API to get parent places instead of using get_prop_value a few times.
* Get rid of "cached" version of some data fetch functions. The dc.Fn() call result is already cached, and/or there is cache for flask endpoints response.

### CODE REVIEW

File: server/routes/place/api.py [hunk 1]
```diff
@@ -566,7 +566,7 @@ def populate_additional_category_data(category):
   all_places = [dcid]
   for t in BAR_CHART_TYPES:
     all_places.extend(raw_page_data.get(t + 'Places', []))
-  names = place_api.get_display_name('^'.join(sorted(all_places)), g.locale)
```
Comment (pradh) [inline server/routes/place/api.py:569]: This is intentionally dropped?

File: server/routes/shared_api/place.py [hunk 2]
```diff
@@ -853,17 +747,15 @@ def api_ranking_chart(dcid):
...
-      parent_type_list = parent.get("types", [])
       parent_place_dcid = parent.get("dcid", "")
-      if parent_type_list:
-        parent_type = parent_type_list[0]
-        # All wanted place types plus continent except CensusZipCodeTabulationArea.
-        if parent_type == "Continent" or (
-            parent_type in ALL_WANTED_PLACE_TYPES and
-            parent_type != "CensusZipCodeTabulationArea"):
-          break
+      parent_type = parent.get('type', '')
```
Comment (pradh) [inline server/routes/shared_api/place.py:753]: super nit: only this line as different quote style, be consistent?

**Review by pradh — APPROVED** [review]
Not super familiar with the code.  LGTM to unblock.

Maybe Jennifer could also look on Monday?

File: server/routes/place/api.py (hunk 1 above)
Comment (shifucun) [reply server/routes/place/api.py:569]: Yes, the function is simplified by only taking a list of dcids.

File: server/routes/shared_api/place.py (hunk 2 above)
Comment (shifucun) [reply server/routes/shared_api/place.py]: fixed

**Pull request merged**